src/common/rsvc.c

```diff
@@ -181,6 +190,14 @@ rsvc_client_process_hint(struct rsvc_client *client,
 				hint->sh_term, hint->sh_rank);
 			return;
 		} else if (hint->sh_term == client->sc_leader_term) {
+			if (ep->ep_rank == hint->sh_rank) {
+				if (client->sc_leader_aliveness < 2) {
+					D_DEBUG(DB_MD, "leader rank bump "
+						"aliveness %u -> 2\n",
+						client->sc_leader_aliveness);
+					client->sc_leader_aliveness = 2;
+				}
+			}
```

Comment:
Indeed. This is where the root cause of the bug is. We can reach the same scenario if the "client->sc_leader_aliveness--" branch is triggered twice for some reason, without any membership changes.